File: ui/src/main/java/com/xmartlabs/base/ui/common/BaseProgressDialog.java
package com.xmartlabs.base.ui.common;

/**
 * Describes a progress dialog.
 * The UI and interactions are entirely up to the developer, providing it respects the following method definitions.
 */
public interface BaseProgressDialog {
  /** Makes the progress dialog visible to the user. **/
  void show();
  /** Hides the progress dialog, so it's no longer visible by the user. */
  void hide();
  /**
   * Destroys the dialog and releases memory.
   * It should be put into hidden state first, if visible.
   * The {@link #show()} method will yield no effect after calling this method.
   */
  void dismiss();
}

File: ui/src/main/java/com/xmartlabs/bigbang/ui/common/BaseProgressDialog.kt
package com.xmartlabs.bigbang.ui.common

/**
 * Describes a progress dialog.
 * The UI and interactions are entirely up to the developer, providing it respects the following method definitions.
 */
interface BaseProgressDialog {
  /** Makes the progress dialog visible to the user.  */
  fun show()

  /** Hides the progress dialog, so it's no longer visible by the user.  */
  fun hide()

  /**
   * Destroys the dialog and releases memory.
   * It should be put into hidden state first, if visible.
   * The [.show] method will yield no effect after calling this method.
   */
  fun dismiss()
}
